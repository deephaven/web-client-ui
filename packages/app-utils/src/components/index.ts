export * from './AppBootstrap';
export * from './AuthBootstrap';
export * from './ConnectionBootstrap';
export * from './FontBootstrap';
export * from './FontsLoaded';
export * from './PluginsBootstrap';
export * from './usePlugins';
export * from './useConnection';
