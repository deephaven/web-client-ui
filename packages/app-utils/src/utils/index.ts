export * from './ConnectUtils';
export * from './createExportLogsContextAction';
