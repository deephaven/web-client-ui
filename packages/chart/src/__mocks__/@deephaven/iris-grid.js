/* eslint-disable class-methods-use-this */
/* eslint-disable max-classes-per-file */
export class Formatter {
  getColumnTypeFormatter() {
    return 'string';
  }
}

export class FormatterUtils {
  static getColumnFormats = jest.fn();

  static getDateTimeFormatterOptions = jest.fn();
}

export class IrisGrid {}

export class TableUtils {
  static getColumnByName = jest.fn();
}

export default IrisGrid;
