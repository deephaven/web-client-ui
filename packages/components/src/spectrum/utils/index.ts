export * from './itemUtils';
export * from './itemWrapperUtils';
export * from './propsUtils';
export * from './themeUtils';
export * from './useRenderNormalizedItem';
export * from './useStringifiedMultiSelection';
export * from './useStringifiedSelection';
