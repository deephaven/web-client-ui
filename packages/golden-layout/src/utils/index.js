import utils from './utils.js';
import EventEmitter from './EventEmitter.js';
import DragListener from './DragListener.js';
import ReactComponentHandler from './ReactComponentHandler.js';
import ConfigMinifier from './ConfigMinifier.js';
import BubblingEvent from './BubblingEvent.js';
import EventHub from './EventHub.js';

export default {
  ...utils,
  EventEmitter,
  DragListener,
  ReactComponentHandler,
  ConfigMinifier,
  BubblingEvent,
  EventHub,
};
