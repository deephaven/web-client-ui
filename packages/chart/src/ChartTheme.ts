import {
  getExpressionRanges,
  resolveCssVariablesInRecord,
} from '@deephaven/components';
import Log from '@deephaven/log';
import { ColorUtils } from '@deephaven/utils';
import chartThemeRaw from './ChartTheme.module.scss';

const log = Log.module('ChartTheme');

export interface ChartTheme {
  paper_bgcolor: string;
  plot_bgcolor: string;
  title_color: string;
  legend_color: string;
  colorway: string;
  gridcolor: string;
  linecolor: string;
  zerolinecolor: string;
  activecolor: string;
  rangebgcolor: string;
  area_color: string;
  trend_color: string;
  line_color: string;
  error_band_line_color: string;
  error_band_fill_color: string;
  ohlc_increasing: string;
  ohlc_decreasing: string;
}

export function defaultChartTheme(): Readonly<ChartTheme> {
  const chartTheme = resolveCssVariablesInRecord(chartThemeRaw);

  // The color normalization in `resolveCssVariablesInRecord` won't work for
  // colorway since it is an array of colors. We need to explicitly normalize
  // each color expression
  chartTheme.colorway = getExpressionRanges(chartTheme.colorway ?? '')
    .map(([start, end]) =>
      ColorUtils.normalizeCssColor(
        chartTheme.colorway.substring(start, end + 1)
      )
    )
    .join(' ');

  log.debug2('Chart theme:', chartThemeRaw);
  log.debug2('Chart theme derived:', chartTheme);

  return Object.freeze({
    paper_bgcolor: chartTheme['paper-bgcolor'],
    plot_bgcolor: chartTheme['plot-bgcolor'],
    title_color: chartTheme['title-color'],
    legend_color: chartTheme['legend-color'],
    colorway: chartTheme.colorway,
    gridcolor: chartTheme.gridcolor,
    linecolor: chartTheme.linecolor,
    zerolinecolor: chartTheme.zerolinecolor,
    activecolor: chartTheme.activecolor,
    rangebgcolor: chartTheme.rangebgcolor,
    area_color: chartTheme['area-color'],
    trend_color: chartTheme['trend-color'],
    line_color: chartTheme['line-color'],
    error_band_line_color: chartTheme['error-band-line-color'],
    error_band_fill_color: chartTheme['error-band-fill-color'],
    ohlc_increasing: chartTheme['ohlc-increasing'],
    ohlc_decreasing: chartTheme['ohlc-decreasing'],
  });
}

export default defaultChartTheme;
