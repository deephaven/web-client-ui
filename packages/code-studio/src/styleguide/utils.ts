import { createElement, useCallback, useEffect, useState } from 'react';
import cl from 'classnames';
import { Item, ItemElement, NormalizedItem } from '@deephaven/components';
import { dh as dhIcons } from '@deephaven/icons';

export const ISOLATED_SECTION_QUERY_CLASS = 'isolated-section-query';
export const SAMPLE_SECTION_CLASS = 'sample-section';

/**
 * Generate a given number of `Item` elements.
 */
export function* generateItemElements(
  start: number,
  end: number
): Generator<ItemElement> {
  const letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
  const len = letters.length;

  for (let i = start; i <= end; i += 1) {
    const charI = i % len;
    let suffix = String(Math.floor(i / len));
    if (suffix === '0') {
      suffix = '';
    }
    const letter = letters[charI];
    const key = `${letter}${suffix}`;
    const content = `${letter.repeat(3)}${suffix}`;

    // eslint-disable-next-line react/no-children-prop
    yield createElement(Item, {
      key,
      textValue: content,
      children: content,
    });
  }
}

/**
 * Generate a given number of NormalizedItems.
 * @param count The number of items to generate
 */
export function* generateNormalizedItems(
  count: number,
  include: { descriptions?: boolean; icons?: boolean } = {}
): Generator<NormalizedItem> {
  const letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
  const len = letters.length;

  const iconKeys = Object.keys(dhIcons);

  for (let i = 0; i < count; i += 1) {
    const charI = i % len;
    let suffix = String(Math.floor(i / len));
    if (suffix === '0') {
      suffix = '';
    }
    const letter = letters[charI];
    const key = `${letter}${suffix}`;

    const icon =
      include.icons === true ? iconKeys[i % iconKeys.length] : undefined;

    const description =
      include.descriptions === true ? `Description ${key}` : undefined;

    const content = icon ?? `${letter.repeat(3)}${suffix}`;

    yield {
      key,
      item: {
        key: (i + 1) * 100,
        content,
        textValue: content,
        description,
        icon,
      },
    };
  }
}

/**
 * Pseudo random number generator with seed so we get reproducible output.
 * This is necessary in order for e2e tests to work.
 */
export function* pseudoRandomWithSeed(
  seed = 1
): Generator<number, void, never> {
  while (true) {
    // eslint-disable-next-line no-param-reassign
    seed = (seed * 16807) % 2147483647;
    yield seed / 2147483647;
  }
}

/**
 * If query string isolatedSection=true, return the current location hash.
 * Otherwise ''
 */
function getIsolatedSectionHash() {
  const isolateSection = window.location.search.includes('isolateSection=true');
  return isolateSection ? window.location.hash.replace(/^#/, '') : '';
}

/**
 * If query string isolatedSection=true, return the current location hash.
 * Otherwise ''. Updates when the hash changes.
 */
export function useIsolateSectionHash(): string {
  const [hash, setHash] = useState(getIsolatedSectionHash);

  useEffect(() => {
    const hashChangeHandler = () => setHash(getIsolatedSectionHash());

    window.addEventListener('hashchange', hashChangeHandler);

    return () => window.removeEventListener('hashchange', hashChangeHandler);
  }, []);

  return hash;
}

/**
 * Returns a callback that will generate pseudo random numbers based on the
 * given seed.
 * @param seed
 */
export function useSeededRandomNumberCallback(seed = 1): () => number {
  const [randomGenerator] = useState(() => pseudoRandomWithSeed(seed));

  return useCallback(() => {
    const result = randomGenerator.next();

    if (result.done === true) {
      throw new Error('Random number generator unexpectedly finished');
    }

    return result.value;
  }, [randomGenerator]);
}

export function getSectionIdFromName(name: string): string {
  return `${SAMPLE_SECTION_CLASS}-${name
    .toLocaleLowerCase()
    .replaceAll(' ', '-')}`;
}

/**
 * Return id and className props for a sample section. Class
 * names generated by this util are used by e2e tests to take snapshots of
 * styleguide sections.
 * @param name Name of the section
 * @param classNames Optional list of class names to include
 */
export function sampleSectionIdAndClasses(
  name: string,
  classNames: string[] = []
): { id: string; className: string } {
  const id = getSectionIdFromName(name);

  const className = cl(SAMPLE_SECTION_CLASS, ...classNames);

  return {
    id,
    className,
  };
}

/**
 * Return id and UNSAFE_className props for a sample section. Class
 * names generated by this util are used by e2e tests to take snapshots of
 * styleguide sections.
 * @param name Name of the section
 * @param classNames Optional list of class names to include
 */
export function sampleSectionIdAndClassesSpectrum(
  name: string,
  classNames: string[] = []
): { id: string; UNSAFE_className: string } {
  const { id, className } = sampleSectionIdAndClasses(name, classNames);

  return {
    id,
    UNSAFE_className: className,
  };
}
