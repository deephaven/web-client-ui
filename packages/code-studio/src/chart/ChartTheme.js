import ChartTheme from './ChartTheme.module.scss';

export default Object.freeze({
  paper_bgcolor: ChartTheme['paper-bgcolor'],
  plot_bgcolor: ChartTheme['plot-bgcolor'],
  title_color: ChartTheme['title-color'],
  colorway: ChartTheme.colorway,
  gridcolor: ChartTheme.gridcolor,
  linecolor: ChartTheme.linecolor,
  zerolinecolor: ChartTheme.zerolinecolor,
  activecolor: ChartTheme.activecolor,
  rangebgcolor: ChartTheme.rangebgcolor,
  area_color: ChartTheme['area-color'],
  trend_color: ChartTheme['trend-color'],
  line_color: ChartTheme['line-color'],
  error_band_line_color: ChartTheme['error-band-line-color'],
  error_band_fill_color: ChartTheme['error-band-fill-color'],
  ohlc_increasing: ChartTheme['ohlc-increasing'],
  ohlc_decreasing: ChartTheme['ohlc-decreasing'],
});
