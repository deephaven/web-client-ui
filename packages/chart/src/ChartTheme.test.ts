/// <reference types="./declaration" />

import { TestUtils } from '@deephaven/utils';
import { resolveCssVariablesInRecord } from '@deephaven/components';
import { defaultChartTheme } from './ChartTheme';
import chartThemeRaw from './ChartTheme.module.scss';

jest.mock('@deephaven/components', () => ({
  ...jest.requireActual('@deephaven/components'),
  resolveCssVariablesInRecord: jest.fn(),
}));

const { asMock } = TestUtils;

const mockChartTheme = new Proxy(
  {},
  { get: (_target, name) => `chartTheme['${String(name)}']` }
);

beforeEach(() => {
  jest.clearAllMocks();
  expect.hasAssertions();

  asMock(resolveCssVariablesInRecord)
    .mockName('resolveCssVariablesInRecord')
    .mockReturnValue(mockChartTheme);
});

describe('defaultChartTheme', () => {
  it('should create the default chart theme', () => {
    const actual = defaultChartTheme();

    expect(resolveCssVariablesInRecord).toHaveBeenCalledWith(chartThemeRaw);
    expect(actual).toMatchSnapshot();
  });
});
