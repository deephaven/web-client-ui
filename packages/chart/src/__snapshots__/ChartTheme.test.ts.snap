// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultChartTheme should create the default chart theme 1`] = `
{
  "activecolor": "chartTheme['activecolor']",
  "area_color": "chartTheme['area-color']",
  "colorway": "chartTheme['colorway']",
  "error_band_fill_color": "chartTheme['error-band-fill-color']",
  "error_band_line_color": "chartTheme['error-band-line-color']",
  "gridcolor": "chartTheme['gridcolor']",
  "line_color": "chartTheme['line-color']",
  "linecolor": "chartTheme['linecolor']",
  "ohlc_decreasing": "chartTheme['ohlc-decreasing']",
  "ohlc_increasing": "chartTheme['ohlc-increasing']",
  "paper_bgcolor": "chartTheme['paper-bgcolor']",
  "plot_bgcolor": "chartTheme['plot-bgcolor']",
  "rangebgcolor": "chartTheme['rangebgcolor']",
  "title_color": "chartTheme['title-color']",
  "trend_color": "chartTheme['trend-color']",
  "zerolinecolor": "chartTheme['zerolinecolor']",
}
`;
